Code/lib/corner/corner.cpp

```diff
@@ -0,0 +1,90 @@
+#include <corner.h>
+
+CornerMerchantry::CornerMerchantry(int trigPin_1, int echoPin_1, int trigPin_2, int echoPin_2, float *right_threshold, float *front_threshold)
+{
+    set_threshold_right = right_threshold;
+    set_threshold_front = front_threshold;
+    this->trigPin_front = trigPin_1;
+    this->echoPin_front = echoPin_1;
+    this->trigPin_side = trigPin_2;
+    this->echoPin_side = echoPin_2;
+    pinMode(trigPin_front, OUTPUT);
+    pinMode(trigPin_side, OUTPUT);
+    pinMode(echoPin_front, INPUT);
+    pinMode(echoPin_side, INPUT);
+}
+float CornerMerchantry::getDistance_front()
+{
+    // eliminate unknown state
+    digitalWrite(trigPin_front, LOW);
+    delayMicroseconds(2);
+    // sending pulse
+    digitalWrite(trigPin_front, HIGH);
+    delayMicroseconds(10);
+    // switching off pulse
+    digitalWrite(trigPin_front, LOW);
+    // reading pulse width and converting to distance
+    float duration = pulseIn(echoPin_front, HIGH);
+    return duration * 0.034 / 2;
+}
+float CornerMerchantry::getDistance_side()
```

Comment:
Since this section is almost similar to `getDistance_front()` why not refactor it to something like this

```cpp
float CornerMerchantry::getDistance_front()
{
    return getDistance(echoPin_front, trigPin_front);
}
float CornerMerchantry::getDistance_side()
{
    return getDistance(echoPin_side, trigPin_side);
}

float getDistance(int echoPin, int trigPin){
    // eliminate unknown state
    digitalWrite(trigPin, LOW);
    delayMicroseconds(2);
    // sending pulse
    digitalWrite(trigPin, HIGH);
    delayMicroseconds(10);
    // switching off pulse
    digitalWrite(trigPin, LOW);
    // reading pulse width and converting to distance
    float duration = pulseIn(echoPin, HIGH);
    return duration * 0.034 / 2;
}
```

Reply:
@SammyOina and @1998-felix